python/sglang/srt/cpu_utils.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+import subprocess
+from collections import defaultdict
+
+
+def parse_lscpu_topology():
+    try:
+        # Get CPU topology: CPU,Core,Socket,Node
+        output = subprocess.check_output(
+            ["lscpu", "-p=CPU,Core,Socket,Node"], text=True
+        )
+    except Exception as e:
+        raise RuntimeError(f"Unexpected error running 'lscpu': {e}")
+
+    # Parse only data lines (skip comments)
+    cpu_info = []
+    for line in output.splitlines():
+        if not line.startswith("#"):
+            cpu, core, socket, node = map(int, line.strip().split(","))
```

Comment:
you could get those information via py-libnuma and pstuil

Reply:
@chunyuan-w